src/scheduler/resv_info.c

```diff
@@ -534,10 +534,10 @@ query_reservations(server_info *sinfo, struct batch_status *resvs)
 			}
 		}
 		resresv_arr[idx] = resresv;
+		resresv_arr[idx + 1] = NULL;
```

Comment:
Alright, I've changed it to use increment operator instead. Since this is a pretty small code change overall, I also squashed the commits into one to avoid another iteration just to do the squash. I hope that's ok.